### ISSUE TITLE
AdminUI update error on timestamp fields using MySQL

### ISSUE THREAD

**Issue by derrickmehaffy** [strapi/strapi#2663]
**Informations**
- **Node.js version**: 10
- **NPM version**: 6
- **Strapi version**: v19
- **Database**: MariaDB 10.3 (MySQL)
- **Operating system**: Ubuntu 18.04
- **(Optional) Link to your Project**: https://github.com/canonn-science/CAPIv2-Strapi

**What is the current behavior?**

Error while updating via console: 
```
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 1, 1 ],
  __knexQueryUid: '5c1267fd-cb9d-45a7-9120-165b040fa5e7',
  sql:
   'select `users-permissions_user`.* from `users-permissions_user` where `users-permissions_user`.`id` = ? limit ?' }
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ 1 ],
  __knexQueryUid: 'c6170a9e-e68f-4cbc-b524-d82c8b739216',
  sql:
   'select `users-permissions_role`.* from `users-permissions_role` where `users-permissions_role`.`id` in (?)' }
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ '1', 1 ],
  __knexQueryUid: '91555762-4f63-4c76-a838-95a14c9b16e5',
  sql:
   'select `excludeclients`.* from `excludeclients` where `excludeclients`.`id` = ? limit ?' }
{ method: 'update',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings:
   [ 1,
     'testclient',
     'Testing 2',
     '2019-01-21T06:42:15.000Z',
     '2019-01-21T06:42:15.000Z',
     1 ],
  __knexQueryUid: 'de9e98ab-192c-4d3f-8571-78c0bca8c830',
  sql:
   'update `excludeclients` set `id` = ?, `version` = ?, `reason` = ?, `created_at` = ?, `updated_at` = ? where `id` = ?' }
[2019-01-21T06:42:21.161Z] error update `excludeclients` set `id` = 1, `version` = 'testclient', `reason` = 'Testing 2', `created_at` = '2019-01-21T06:42:15.000Z', `updated_at` = '2019-01-21T06:42:15.000Z' where `id` = 1 - ER_TRUNCATED_WRONG_VALUE: Incorrect datetime value: '2019-01-21T06:42:15.000Z' for column `capiv2_example`.`excludeclients`.`created_at` at row 1
```

It appears you can create new entry it works fine, however attempting to update the time format is wrong. @soupette just a heads up with MySQL the update format has to be `YYYY-MM-DD HH:MM:SS` in the raw queries. 

**Steps to reproduce the problem**

Spin up a v19 app, create a model, add an entry, then try and update it.

**What is the expected behavior?**

Updates should work :stuck_out_tongue: 

**Suggested solutions**

To be honest I seriously think the dates should not be formatted via whatever is being used (moment.js ?) Even the API responds with a different date format than what it actually accepts:

![image](https://user-images.githubusercontent.com/8593673/51457550-09582f80-1d0f-11e9-8e64-287c2a4f576b.png)

Notice in that screenshot I am sending `"updated_at": "2019-01-21 06:53:15"` which is the only way to get put requests work. Whatever format that is being used that generates `"updated_at": "2019-01-21T06:42:15.000Z"` this format should be removed in my opinion

**Comment by suhaboncukcu** [comment]
@derrickmehaffy I think removing this format could lead some problems while handling multiple timezones. May be a change on the accepted field could serve better.

**Comment by derrickmehaffy** [comment]
@suhaboncukcu typically when you are dealing with an API like this, you standardize the timezone, IE you only use and accept a single timezone then force your frontend to convert as needed depending on the location. Usually UTC is what is used.

**Comment by suhaboncukcu** [comment]
@derrickmehaffy I guess Django uses a serializer for these kind of cases right? I am willing to open a PR if we find a general case solution.

**Comment by derrickmehaffy** [comment]
We need to figure out a solution for everyone, waiting on input from @Aurelsicoko @lauriejim and @soupette 

There have been a ton of issues dealing with dates and we need to come to common terms with proper formats, and both backend/frontend with how they should be handled.

**Comment by odelucca** [comment]
I agree with @derrickmehaffy

Almost every API that I've made or saw uses UTC as default. It isn't a good practice use a formatted date in the database.

**Comment by suhaboncukcu** [comment]
So, what about a "presentation and transformation layer" for an all in one solution? Making dates fixed in the timezone of application config and get & serve according to the layer settings. I guess something like this also could be used for different response standards like HAL.

**Comment by derrickmehaffy** [comment]
@suhaboncukcu translation and conversion should be done on the client side not the server side. Overly complicates the server IMO. client gets UTC no matter what from the server, so you always know the date and time is clear and accurate.

After that there is libraries for just about every language under the sun to convert UTC to local timezone.

**Comment by derrickmehaffy** [comment]
It's the same method that MySQL has been doing for the past 24 years or so.

**Comment by suhaboncukcu** [comment]
Oh, I was talking about something like thephpleague's fractal package. Not specifically for date :)  Sorry for non-clear comment. 

But I'm also not sold about removing the timezone indicator from the returning date completely since lots of different frameworks provide an option to change default timezone. There must be some use cases for people out there. Even MySql's timezone setting is config based. 

There might be a pseudo date type as "timezone aware date" to serve dates with timezones though.

**Comment by derrickmehaffy** [comment]
@suhaboncukcu sure there is, doesn't mean it should be used. There is no major drawback from using one standard timezone with the exception that each of your frontend needs to be aware of that. The downside to running it server side is the potential not just break one frontend, but all of them. The goal of strapi is to be as agnostic as possible, thus you want certain standards in place such as a solid and clean date, time, and datetime system in place. 

At it's core Strapi is not just a headless CMS but is an API framework. Look at any other major API, they all use UTC. Why because that is **the standard** of date and time that all timezones are based on. :stuck_out_tongue:

**Comment by derrickmehaffy** [comment]
conversion of timezones at the client level is far easier than conversion on the server side.

**Comment by derrickmehaffy** [comment]
Here is a good read if you have a few minutes: https://zachholman.com/talk/utc-is-enough-for-everyone-right

**Comment by derrickmehaffy** [comment]
@lauriejim The bug is back

**Comment by jaime8111** [comment]
I've start a new strapi project with MondoDB instead of MySQL and works perfect. For me is OK, i'm begining with the CMS and I don't mind to change from MySQL to MongoDB, but you have to fix this bug for other users. It looks very important.

**Comment by lauriejim** [comment]
Can you test with last version of Strapi please.

**Comment by lauriejim** [comment]
I just tried and t work on last version of Strapi.

**Comment by Slauta** [comment]
I have this bag in strapi version v3.0.0-alpha.26.1.
I created project with mongo db and migrated to mysql.

**Comment by lauriejim** [comment]
Hello @Slauta !
The generated code between a Mongo and MySQL application is not the same.
So issues will appear.

**Comment by petre2dor** [comment]
This exact bug is back. I used v3.0.0-beta.3 and upgraded to v3.0.0-beta.6 and it affects both versions. Please re-open this issue.

**Comment by derrickmehaffy** [comment]
@petre2dor the issue is already known about and there is a pending PR for this.

**Comment by lauriejim** [comment]
I just tested it with a fresh new app the `beta.7` and it works well.
I'm able to update a `date` input.

**Comment by lauriejim** [comment]
Can you please @petre2dor tell me steps to reproduce your issue please.

**Comment by petre2dor** [comment]
@lauriejim sure:
1. installed strapi beta.7 using mysql as DB
2. create a new user
3. trying to save the changes to the user (update) returns this error:

> update `users-permissions_user` set `created_at` = '2019-06-27T14:05:07.000Z', `blocked` = false, `resetPasswordToken` = NULL, `provider` = 'local', `role` = 1, `username` = 'test', `updated_at` = '2019-06-27T14:05:07.000Z', `password` = '$2a$10$L2Fbj5Lh3w2I9T7r8ZzgWehvYtzIzKkZKd1FYxP0Q7TMg8W5LHGXO', `email` = 'me@myemailaddress.nope', `confirmed` = false where `id` = 1 - ER_TRUNCATED_WRONG_VALUE: Incorrect datetime value: '2019-06-27T14:05:07.000Z' for column `test`.`users-permissions_user`.`created_at` at row 1


Creating a project using `--quickstart` works fain with the steps above.
I see that the PR that is supposed to fix this is still not merged, so I'll wait for it and upgrade.

**Comment by lauriejim** [comment]
@petre2dor Yes you are right!
Can you please open a new issue about that please.
Appreciate.

**Comment by petre2dor** [comment]
@lauriejim someone already re-opened this issue #1579. I would say there is no need for a new one.

**Comment by lauriejim** [comment]
OKay thank you for the update.
